### PR TITLE
[SPARK-9967][SPARK-10099][STREAMING] Renamed conf spark.streaming.backpressure.{enable-->enabled} and fixed deprecated annotations

### PR DESCRIPTION
**Pull request by tdas**
Small changes
- Renamed conf spark.streaming.backpressure.{enable --> enabled}
- Change Java Deprecated annotations to Scala deprecated annotation with more information.

### CODE REVIEW

File: streaming/src/main/scala/org/apache/spark/streaming/scheduler/RateController.scala [hunk 1]
```diff
@@ -76,9 +76,9 @@ private[streaming] abstract class RateController(val streamUID: Int, rateEstimat
     val elements = batchCompleted.batchInfo.streamIdToInputInfo
 
     for {
-      processingEnd <- batchCompleted.batchInfo.processingEndTime;
-      workDelay <- batchCompleted.batchInfo.processingDelay;
-      waitDelay <- batchCompleted.batchInfo.schedulingDelay;
+      processingEnd <- batchCompleted.batchInfo.processingEndTime
```
Comment (tdas) [inline streaming/src/main/scala/org/apache/spark/streaming/scheduler/RateController.scala]: The semicolons where unnecessary and was being flagged by IntelliJ

**Comment by andrewor14** [comment]
yes, please call it `enabled` instead

**Comment by tdas** [comment]
LGTY?

**Comment by andrewor14** [comment]
yeah feel free to merge (FFTM)

**Comment by srowen** [comment]
Oh good, I was just about to. LGTM.

**Comment by tdas** [comment]
The last failures was unrelated ConnectionManagerSuite (which is known to be flaky) and the currently running core tests have all passed I am merging this to master and 1.5. Thanks @andrewor14 and @srowen .

File: streaming/src/main/scala/org/apache/spark/streaming/scheduler/RateController.scala [hunk 2]
```diff
@@ -86,5 +86,5 @@ private[streaming] abstract class RateController(val streamUID: Int, rateEstimat
 
 object RateController {
   def isBackPressureEnabled(conf: SparkConf): Boolean =
-    conf.getBoolean("spark.streaming.backpressure.enable", false)
+    conf.getBoolean("spark.streaming.backpressure.enabled", false)
```
Comment (srowen) [inline streaming/src/main/scala/org/apache/spark/streaming/scheduler/RateController.scala]: The idea here was that this flag hadn't been 'released' yet so OK to change the name without keeping the old one?

The two changes in the PR were pretty unrelated right? still I can't imagine someone would really care about back-porting one and not be able to consume the other as they're minor.

Comment (tdas) [inline streaming/src/main/scala/org/apache/spark/streaming/scheduler/RateController.scala]: Yes, for both. That's why went ahead with squashing both into one.

**Pull request closed without merging**